### PR TITLE
feat(frontend): move to other grid

### PR DESCRIPTION
**Pull request by fatonramadani**
https://user-images.githubusercontent.com/456655/221056600-288dae0a-d6d0-4129-ae4c-9966a4132886.mov

### CODE REVIEW

**Comment by fatonramadani** [comment]
@rubenfiszel Migrated to new architecture

File: frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+		$selectedComponent = id
+	}
+
+	$: availableGrids = listAllSubGrids($app)
+
+	function onMove() {
+		if (component) {
+			const [targetId, targetSubGridIndex] = (
+				document.getElementById('move-to-grid') as HTMLSelectElement
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte]: why not bind the value of the select ?

Comment (rubenfiszel) [reply frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte]: https://svelte.dev/tutorial/select-bindings

File: frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+				}
+
+				const data: AppComponent = JSON.parse(JSON.stringify(node.data))
+				$dirtyStore = true
+
+				deleteGridItem($app, component, parent)
+
+				const newComponent = createNewGridItem($app.grid, component.id, data)
+				$app.grid = [...$app.grid, newComponent]
+				$selectedComponent = newComponent.id
```
Comment (rubenfiszel) [inline frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte]: lots of copy pasted with moveComponent, probably some aspect can be factorized

File: frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte (hunk 1 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte]: Done!

File: frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte (hunk 2 above)
Comment (fatonramadani) [reply frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte]: Done!

**Pull request merged**